run_task_and_await_result.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash 
+function usage() {
+    echo "CLUSTER must be set to desired ecs-cluster"
+    echo "TASK_DEF must be set to desired task-definition:revision number"
+    echo "Example:"
+    echo "$ CLUSTER=data-platform-ecs-cluster-green TASK_DEF=hello-world-task:6 ./run_task_and_await_result.sh"
+    exit 1
+}
+
+#check requirements
+function require {
+    command -v $1 > /dev/null 2>&1 || {
+        echo "Some of the required software is not installed:"
+        echo "    please install $1" >&2;
+        exit 1;
+    }
+}
+
+require aws
+require jq
+
+if [[ -z "$CLUSTER" ]] || [[ -z "$TASK_DEF" ]]; then
+    usage
+fi
+
+TASK_ARN=$(aws ecs run-task --cluster $CLUSTER --task-definition $TASK_DEF | jq -r .tasks[0].containers[0].taskArn)
+echo "TASK_ARN is $TASK_ARN"
+
+function wait_for_result { 
+    while [[ "$TASK_STATUS" != "STOPPED" ]]; do 
+        sleep 3
+        TASK_JSON=$(aws ecs describe-tasks --cluster $CLUSTER --tasks $TASK_ARN)
+        TASK_STATUS=$(jq -r .tasks[0].containers[0].lastStatus <<< $TASK_JSON)
+        echo "Current task status is $TASK_STATUS"
+    done
+
+    TASK_EXIT_CODE=$(jq -r .tasks[0].containers[0].exitCode <<< $TASK_JSON)
+    echo "Task staus eventually reached $TASK_STATUS"
+}
+
+wait_for_result
+
+echo "Exit code for container $(jq .tasks[0].containers[0].name <<< $TASK_JSON) was $TASK_EXIT_CODE"
```

Comment:
Hmm slightly confusing that `TASK_JSON` is only assigned to within that function. Maybe set it to empty first then `global TASK_JSON` (I think the syntax was) from inside the function. But NBD :)

Reply:
fair enough, i'm all about that clarity

Reply:
maybe i'll extract the status refresh to a separate function all together

Reply:
Actually, my bash isn't eloquent away to think of a better way to make it more readable. I think it's not too bad as it is but if it really becomes a problem then I can  refactor a little